src/nvim/eval.c

```diff
@@ -10571,18 +10594,13 @@ static void f_getregtype(typval_T *argvars, typval_T *rettv)
   if (regname == 0)
     regname = '"';
 
-  buf[0] = NUL;
-  buf[1] = NUL;
-  switch (get_reg_type(regname, &reglen)) {
-  case MLINE: buf[0] = 'V'; break;
-  case MCHAR: buf[0] = 'v'; break;
-  case MBLOCK:
-    buf[0] = Ctrl_V;
-    sprintf((char *)buf + 1, "%" PRId64, (int64_t)(reglen + 1));
-    break;
-  }
+  colnr_T reglen = 0;
+  char buf[NUMBUFLEN + 2];
+  char_u reg_type = get_reg_type(regname, &reglen);
+  format_reg_type(reg_type, reglen, buf, ARRAY_SIZE(buf));
+
   rettv->v_type = VAR_STRING;
-  rettv->vval.v_string = vim_strsave(buf);
+  rettv->vval.v_string = vim_strsave((char_u *)buf);
```

Comment:
If you have buffer updated to `char` it is also logical to update `vim_strsave((char_u *)` to `(char_u *) xstrdup(`: when `char_u` is removed `vim_strsave` is no longer needed (it is actually doing nothing more then simply calling `xstrdup` in turn, but has different types; `xstrdup` is newer).